### PR TITLE
Add deprecation notice for CentOS

### PR DESCRIPTION
**Pull request by tianon**
Closes https://github.com/docker-library/official-images/issues/11679

### CODE REVIEW

File: centos/deprecated.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This image is no longer supported/maintained (last updated September 15, 2021, [docker-library/official-images#10304](https://github.com/docker-library/official-images/pull/10304); see also https://www.centos.org/centos-linux-eol/). Please adjust your usage accordingly.
```
Comment (yosifkit) [inline centos/deprecated.md:1]: Should this point to the earlier PRs that actually changed `7.x` (https://github.com/docker-library/official-images/pull/9082 and https://github.com/docker-library/official-images/pull/9102 for `ppc64le` and https://github.com/docker-library/official-images/pull/5146 for `arm32v7` :see_no_evil:)?

Comment (tianon) [reply centos/deprecated.md:1]: Yeah, it's a bit complicated!  This PR is the last time the image maintainer updated any tag of the image, but there's one slightly newer where you removed the EOL versions (which is what the PR I linked to was updating) where these are the most recent PRs to update an ostensibly "supported" version. :weary:

Maybe we should change the wording to make it clear this is the last time the maintainer updated any part of the image, not necessarily the supported tags?  Maybe we should instead drop this bit of the message entirely?  Not sure.

Comment (tianon) [reply centos/deprecated.md]: Ok, updated with some alternative wording for your consideration. :bow:

**Comment by tianon** [comment]
For future readers of this PR, I'd like to be absolutely clear that I'd love to revert this! :heart:

What's necessary for that to happen is a proper update to the supported versions of CentOS, and a commitment to keep them appropriately updated in the future (on a timely schedule -- we recommend once a month at the very most). :bow:

**Review by yosifkit — APPROVED** [review]

**Pull request merged**